internal/rotate.go

```diff
@@ -196,19 +196,26 @@ func (h *CertRotationHandler) RestartServices() {
 	if err != nil {
 		log.Fatalf("Unable to connect to DBUS for service restarts: %s", err)
 	}
-
-	for _, svc := range []string{"aktualizr-lite.service", "fioconfig.service"} {
+	services := []string{"aktualizr-lite.service", "fioconfig.service"}
+	units, err := con.ListUnitsByNamesContext(ctx, services)
+	if err != nil {
+		log.Fatalf("Unable to query status of units: %s", err)
+	}
+	for _, unit := range units {
 		restartChan := make(chan string)
-		_, err = con.RestartUnitContext(ctx, svc, "replace", restartChan)
+		if unit.ActiveState == "inactive" || unit.SubState == "dead" {
+			continue
```

Comment:
I think i'd rather restart it anyway. If this is the fioconfig daemon it needs to restart or it won't be able to talk to the server anymore.

Reply:
So the discussion was indeed actually only concerning `aklite` so I will amend that to be the case.